Permanent/ViewControllers/TagManagementViewController.swift

```diff
@@ -0,0 +1,186 @@
+        
+        archiveTitleNameLabel.font = Text.style35.font
+        archiveTitleNameLabel.textColor = .darkBlue
+        
+        archiveTitleTagsNbrLabel.font = Text.style34.font
+        archiveTitleTagsNbrLabel.textColor = .lightGray
+        
+        if let archiveName = AuthenticationManager.shared.session?.selectedArchive?.fullName {
+            archiveTitleNameLabel.text = "The \(archiveName) Archive Tags"
+            updateTagsNumber()
+        } else {
+            archiveTitleNameLabel.text = "Loading..."
+        }
+        
+        searchTags.updateHeight(height: 40, radius: 2)
+        if #available(iOS 13.0, *) {
+            searchTags.searchTextField.font = Text.style39.font
+        }
+        searchTags.setPlaceholderTextColor(.lightGray)
+        searchTags.tintColor = .lightGray
+        searchTags.searchTextPositionAdjustment = UIOffset(horizontal: 8, vertical: 0)
+        searchTags.barTintColor = .white
+        searchTags.backgroundColor = .white
+        
+        addButton.delegate = self
+        searchTags.delegate = self
+        
+        addButton.isHidden = true
+    }
+    
+    func initCollectionView() {
+        collectionView.register(ArchiveSettingsTagCollectionViewCell.nib(), forCellWithReuseIdentifier: ArchiveSettingsTagCollectionViewCell.identifier)
+        collectionView.register(ArchiveSettingsTagsHeaderCollectionView.nib(), forSupplementaryViewOfKind: ArchiveSettingsTagsHeaderCollectionView.kind, withReuseIdentifier: ArchiveSettingsTagsHeaderCollectionView.identifier)
+        
+        collectionView.backgroundColor = .clear
+        collectionView.indicatorStyle = .white
+        
+        let flowLayout = UICollectionViewFlowLayout()
+        collectionView.collectionViewLayout = flowLayout
+    }
+    
+    func updateTagsNumber() {
+        guard let tagsNumber = viewModel?.sortedTags.count else {
+            archiveTitleTagsNbrLabel.text = nil
+            return
+        }
+        
+        archiveTitleTagsNbrLabel.text = "(\(String(tagsNumber)))"
+    }
+}
+
+// MARK: - FABView Delegate
+extension TagManagementViewController: FABViewDelegate {
+    func didTap() {
+        guard let actionSheet = UIViewController.create(
+            withIdentifier: .fabActionSheet,
+            from: .main
+        ) as? FABActionSheet else {
+            showAlert(title: .error, message: .errorMessage)
+            return
+        }
+
+        //actionSheet.delegate = self
+        navigationController?.display(viewController: actionSheet, modally: true)
+    }
+}
+
+// MARK: - SearchBar Delegate
+extension TagManagementViewController: UISearchBarDelegate {
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+
+        let currentTagVO: [TagVO] = viewModel?.tags.map({ (item) -> TagVO in
+            return item
+        }) ?? []
+
+        viewModel?.sortedTags = searchText.isEmpty ? currentTagVO  : currentTagVO.filter({ (tag) -> Bool in
```

Comment:
viewModel should have a function called viewModel.searchTags(withText text: String) and all of this logic should be implemented in the viewModel.

Implementing it this way will make it testable by unit tests